### PR TITLE
docs: add changes to v16 migration

### LINKED ISSUE #3821: Docs: Add migration guide for version 16 release
### Information

Includes:

- Minimum requirements
- Breaking changes
- Notable features
- Provided Migrations

### Documentation page

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #3821

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/migration/v16.md [hunk 1]
```diff
@@ -21,6 +21,169 @@ Version 16 has the minimum version requirements:
 
 ## Breaking changes
 
-### TBD
 
+## Breaking changes
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/migration/v16.md:25]: The `Breaking changes` title is added twice

File: projects/ngrx.io/content/guide/migration/v16.md [hunk 2]
```diff
@@ -21,6 +21,169 @@ Version 16 has the minimum version requirements:
...
+
+### @ngrx/store
+
+### Preserve the event name case with createActionGroup
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/migration/v16.md:29]:
```suggestion
#### Preserve the event name case with createActionGroup
```

File: projects/ngrx.io/content/guide/migration/v16.md [hunk 3]
```diff
@@ -21,6 +21,169 @@ Version 16 has the minimum version requirements:
...
+// generated actions:
+const {
+  logInSuccess,
+  loginFailure,
+  logoutSuccess,
+  logoutFailure,
+} = authApiActions;
+```
+
+### Strongly typed `createFeature` selectors
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/migration/v16.md:81]:
```suggestion
#### Strongly typed `createFeature` selectors
```

File: projects/ngrx.io/content/guide/migration/v16.md [hunk 4]
```diff
@@ -21,6 +21,169 @@ Version 16 has the minimum version requirements:
...
+  name: 'counter',
+  reducer: createReducer({ count: 0 }),
+});
+
+counterFeature.selectCount.projector
+// ^ type: (featureState: { count: number; }) => number
+```
+
+
+#### Remove `createFeature` Signature with Root State
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/migration/v16.md:114]: just to keep it consistent with titles from other sections

```suggestion
#### Remove `createFeature` signature with root state
```

File: projects/ngrx.io/content/guide/migration/v16.md (hunk 1 above)
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/migration/v16.md:25]: Woops, seems like there are some issues here 😅
Thanks for the review, I'll get these tomorrow

Comment (timdeschryver) [reply projects/ngrx.io/content/guide/migration/v16.md]: It should be better now

**Review by markostanimirovic — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**